### PR TITLE
Add semantic conventions for Resource and Trace attributes

### LINKED ISSUE #376: Semantic Convention Generator
*Is your feature request related to a problem?*
This is an issue that covers similar ideas to issue #297 

Currently there is no semantic convention generator to generate constants

*Describe the solution you'd like*

A semantic convention generator should be added as detailed in the following specifications:

https://github.com/open-telemetry/build-tools/tree/main/semantic-conventions

*Describe alternatives you've considered*
No other alternatives. This is necessary for GA release

### PR DESCRIPTION
**Pull request by lalex**
This PR adds semantic conventions for Resource and Trace attributes.

Fixes #376.

As spec [says](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/overview.md#semantic-conventions), semconv is a candidate for the separated package. So generated classes are in their own namespace `\OpenTelemetry\SemConv`.

There is a script to generate PHP classes from yaml semantic conventions files, (usage `SEMCONV_VERSION=1.8.0 make semconv`).

Deprected class `\OpenTelemetry\SDK\Resource\ResourceConstants`.

### CODE REVIEW

File: script/semantic-conventions/templates/AttributeValues.php.j2
Comment (bobstrecansky) [inline script/semantic-conventions/templates/AttributeValues.php.j2:1]: should we perhaps add this to the `.gitignore` file, or are we purposefully vendoring these into a commit?

**Review by bobstrecansky — COMMENTED** [review]
One open question about committing generated files, but otherwise I very much like the direction.  Thanks for your hard work on this @lalex !

File: script/semantic-conventions/templates/AttributeValues.php.j2
Comment (lalex) [reply script/semantic-conventions/templates/AttributeValues.php.j2:1]: This is a template for the [code generation](https://github.com/open-telemetry/build-tools/tree/main/semantic-conventions#code-generator). It will be used every time the semantic conventions are updated in [spec repo](https://github.com/open-telemetry/opentelemetry-specification/tree/main/semantic_conventions) to generate an updated PHP code.

A workflow will be like this:
- checkout opentelemetry-php repo
- install composer dev requirements
- make semconv
- commit and push an updated generated files

Comment (bobstrecansky) [reply script/semantic-conventions/templates/AttributeValues.php.j2:1]: Figured as much, just wanted to verify.  We should probably add that to the README.md as well.  Would you like to include that in this PR or a subsequent PR?

Comment (lalex) [reply script/semantic-conventions/templates/AttributeValues.php.j2:1]: Done. I've added some lines in README.md.

**Review by bobstrecansky — APPROVED** [review]

**Comment by brettmc** [comment]
@lalex  - it looks like we're still waiting on CLA signing before this can be merged.

**Comment by bobstrecansky** [comment]
He has definitely signed the CLA before; wonder if easyCLA is being annoying.

**Pull request merged**
